bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/catalog/Simbad.scala

```diff
@@ -29,47 +29,54 @@ object Simbad extends Catalog with App {
+    kvs                               = table.fields.zip(row)
 
     // Local find function
-    str = (s:String) => kvs.find(_._1.ucd.exists(_.toLowerCase == s.toLowerCase)).map(_._2)
-    num = (s:String) => str(s).flatMap(_.toDoubleOption)
+    str                               = (s:String) => kvs.find(_._1.ucd.exists(_.toLowerCase == s.toLowerCase)).map(_._2)
+    num                               = (s:String) => str(s).flatMap(_.toDoubleOption)
+    // Find magnitudes checking ucd and field name
+    magStr                            = (b: String, s:String) => kvs.find(i => i._1.ucd.exists(_.toLowerCase == s.toLowerCase) && i._1.id.endsWith(b)).map(_._2)
+    magNum                            = (b: MagnitudeBand, s:String) => magStr(b.name, s).flatMap(_.toDoubleOption)
 
     // Switch to Option here to pull out data
-    epoch <- vot.definitions.map(_.cooSys.epoch).map {
-      case "J2000" => I.CoordinatesEpoch.J_2000
-      case s       => I.CoordinatesEpoch.forName(s)
-    }
-    name <- str("meta.id;meta.main")
-    ra <- num("pos.eq.ra;meta.main")
-    dec <- num("pos.eq.dec;meta.main")
+    epoch                            <- vot.definitions.map(_.cooSys.epoch).map {
+                                         case "J2000" => I.CoordinatesEpoch.J_2000
+                                         case s       => I.CoordinatesEpoch.forName(s)
+                                       }
+    name                             <- str("meta.id;meta.main")
+    ra                               <- num("pos.eq.ra;meta.main")
+    dec                              <- num("pos.eq.dec;meta.main")
 
     // Mags get pulled out into a list
-    mags = for {
-      (k, Some(v)) <- Map(
-        MagnitudeBand.U -> num("phot.mag;em.opt.U"),
-        MagnitudeBand.V -> num("phot.mag;em.opt.V"),
-        MagnitudeBand.B -> num("phot.mag;em.opt.B"),
-        MagnitudeBand.R -> num("phot.mag;em.opt.R"),
-        MagnitudeBand.J -> num("phot.mag;em.ir.J"),
-        MagnitudeBand.H -> num("phot.mag;em.ir.H"),
-        MagnitudeBand.K -> num("phot.mag;em.ir.K"))
-    // TODO: more passbands
-    } yield new Magnitude(v, k, MagnitudeSystem.VEGA) // TODO
+    mags                              = for {
+                                           (k, Some(v)) <- Map(
+                                             MagnitudeBand._u -> magNum(MagnitudeBand._u, "phot.mag;em.opt.U"),
+                                             MagnitudeBand._g -> magNum(MagnitudeBand._g, "phot.mag;em.opt.B"),
+                                             MagnitudeBand._r -> magNum(MagnitudeBand._r, "phot.mag;em.opt.R"),
+                                             MagnitudeBand._i -> magNum(MagnitudeBand._i, "phot.mag;em.opt.I"),
+                                             MagnitudeBand._z -> magNum(MagnitudeBand._z, "phot.mag;em.opt.I"),
+                                             MagnitudeBand.U  -> magNum(MagnitudeBand.U, "phot.mag;em.opt.U"),
```

Comment:
Yeah, that's ugly. On my sample both magnitudes have the same "UCD" `phot.mag;em.opt.U` but the field id is different just in the case. for U it is `FLUX_U` and for u is `FLEX_u`. See the `magStr` method
I hope to add Simbad parsing logic to the catalog bundle and replace this part at some moment in time